src/backend/distributed/commands/function.c

```diff
@@ -192,6 +202,26 @@ create_distributed_function(PG_FUNCTION_ARGS)
 
 	ObjectAddressSet(functionAddress, ProcedureRelationId, funcOid);
 
+	DistObjectCacheEntry *cacheEntry =
+		LookupDistObjectCacheEntry(ProcedureRelationId, funcOid, InvalidOid);
+
+	if (cacheEntry != NULL && IsObjectDistributed(&functionAddress) &&
```

Comment:
We can pass the functionAddress to the IsCreateDistributedFunctionCallIdempotent and include cacheEntry != NULL && IsObjectDistributed(&functionAddress) checks as a part of that function as well.